test/ResourceMatcher.test.js

```diff
@@ -598,6 +598,157 @@ describe("testResourceMatcher", () => {
         // translation of the plural adds or subtracts plural categories creating false matches
         expect(!actual).toBeTruthy();
     });
+});
+
+describe("resource-snake-case", () => {
+    test("defines the built-in declarative resource-matcher rule", () => {
+        const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
+
+        expect(rule.name).toBe("resource-snake-case");
+        expect(rule).toBeInstanceOf(ResourceMatcher);
+    });
+
+    test("sets the rule severity to 'error'", () => {
+        const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
+
+        expect(rule.severity).toBe("error");
+    });
+
+    test("sets the rule regexps to snake case variations", () => {
+        const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
 
+        expect(rule.re).toEqual(expect.arrayContaining([
+            /^\s*[a-z0-9]+(_[a-z0-9]+)+\s*$/dgu, // snake_case
```

Comment:
The unit tests should not really know or even care how it is implemented inside.

Reply:
Fixed